### PR TITLE
[backport/2.2] Update k8s_scale to use module_utils refactored code

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Update `k8s_scale` to use `module_utils` refactored code

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

k8s_scale

### CODE REVIEW

File: plugins/modules/k8s_scale.py [hunk 1]
```diff
@@ -194,7 +202,10 @@ def execute_module(
...
         return_attributes["duration"] = 0
 
-    resource = k8s_ansible_mixin.find_resource(kind, api_version, fail=True)
+    svc = K8sService(client, module)
+    resource = svc.find_resource(
+        module.params.get("kind"), module.params.get("api_version"), fail=True
+    )
 
     from ansible_collections.kubernetes.core.plugins.module_utils.common import (
         NotFoundError,
```
Comment (gravesm) [inline plugins/modules/k8s_scale.py:211]: This should be removed and we should import this directly from kubernetes so we remove the dependency on common.py

File: plugins/modules/k8s_scale.py [hunk 2]
```diff
@@ -277,14 +287,7 @@ def _continue_or_exit(warn):
                     result = resource.patch(existing.to_dict()).to_dict()
                 else:
                     result = scale(
```
Comment (gravesm) [inline plugins/modules/k8s_scale.py:289]: Wrap this scale() call in try/except looking for a CoreException and if it catches one call module.fail_json

Comment (alinabuzachis) [reply plugins/modules/k8s_scale.py]: Done!

**Comment by alinabuzachis** [comment]
recheck

**Comment by Akasurde** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

File: plugins/modules/k8s_scale.py [hunk 3]
```diff
@@ -163,17 +187,9 @@
...
-
-    definition = k8s_ansible_mixin.resource_definitions[0]
-
-    name = definition["metadata"]["name"]
-    namespace = definition["metadata"].get("namespace")
-    api_version = definition["apiVersion"]
-    kind = definition["kind"]
+def execute_module(client, module):
+    namespace = module.params.get("namespace")
+    name = module.params.get("name")
```
Comment (gravesm) [inline plugins/modules/k8s_scale.py:192]: This needs to call `create_definitions()` from `resource.py` first, and then get `namespace`, `name`, `api_version` and `kind` from the returned definition.

**Review by gravesm — APPROVED** [review]

**Pull request merged**